pkg/shared/families/malware/yara/util/yara_util_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/common"
+)
+
+func Test_extractDetectedMalware(t *testing.T) {
+	type args struct {
+		words []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *common.DetectedMalware
+	}{
+		{
+			name: "words len = 0",
+			args: args{
+				words: []string{},
+			},
+			want: nil,
+		},
+		{
+			name: "words are exist",
+			args: args{
+				words: []string{
+					"vmdetect",
+					"author=\"nex\",description=\"Possibly employs anti-virtualization techniques\"",
```

Comment:
you can use "raw" strings via backticks to avoid all the escaping in these cases:
```
`author="nex",description="..."`
```